lib/surgex/parser/parsers/contain_parser.ex

```diff
@@ -0,0 +1,15 @@
+defmodule Surgex.Parser.ContainParser do
+  @moduledoc """
+  Checks if the given parameter's value is on the list of allowed values
+  """
+
+  def call(nil, _allowed_values), do: {:error, :value_is_not_allowed}
+  def call(_, []), do: {:error, :value_is_not_allowed}
+  def call(input, allowed_values) when is_list(allowed_values) do
+    case Enum.member?(allowed_values, input) do
```

Comment:
You could do this inside a guard (which is always better & more efficient). 

```elixir
def call(input, allowed_values) when is_list(allowed_values) and input in allowed_values, do: {:ok, input}
def call(_input, _allowed_values), do: {:error, :invalid_value}
```

(I'd go with `invalid_value` to stick with "invalid" prefix format, since your error code doesn't bring any more description compared to that.)

(This also solves the dilemma about checking list length. It's redundant when you match with `when .. in ...`.)